test/node-report/test-api-uvhandles.js

```diff
@@ -0,0 +1,130 @@
+        fs.unwatchFile(__filename);
+        udp_socket.close();
+        process.removeListener('disconnect', exit);
+      });
+    });
+    req.resume();
+  });
+  server.listen(() => {
+    const data = { pid: child_process.pid,
+                   tcp_address: server.address(),
+                   udp_address: udp_socket.address(),
+                   skip_fs_watch: (watcher === undefined ?
+                     'fs.watch() unavailable' :
+                     false) };
+    process.send(data);
+    http.get({ port: server.address().port });
+  });
+} else {
+  const helper = require('./helper.js');
+  const fork = require('child_process').fork;
+  const assert = require('assert');
+
+  const options = { encoding: 'utf8', silent: true };
+  const child = fork('--experimental-report', [__filename, 'child'], options);
+  let stderr = '';
+  child.stderr.on('data', (chunk) => { stderr += chunk; });
+  let stdout = '';
+  const signal_msg = 'Found messages in stderr unexpectedly: ';
+  const report_msg = 'Report files were written: unexpectedly';
+  child.stdout.on('data', (chunk) => { stdout += chunk; });
+  child.on('exit', (code, signal) => {
+    assert.deepStrictEqual(code, 0, 'Process exited unexpectedly with code' +
+                           `${code}`);
+    assert.deepStrictEqual(signal, null, 'Process should have exited cleanly,' +
+                            ` but did not: ${signal}`);
+    assert.deepStrictEqual(stderr, '', signal_msg, stderr);
+    const reports = helper.findReports(child.pid);
+    assert.deepStrictEqual(reports, [], report_msg, reports);
+
+    const report = JSON.parse(stdout);
+    let fs = 0;
+    let poll = 0;
+    let process = 0;
+    let timer = 0;
+    let pipe = 0;
+    let tcp = 0;
+    let udp = 0;
+    const fs_msg = 'fs_event not found';
+    const poll_msg = 'poll_event not found';
+    const process_msg = 'process event not found';
+    const timer_msg = 'timer event not found';
+    const pipe_msg = 'pipe event not found';
+    const tcp_msg = 'tcp event not found';
+    const udp_msg = 'udp event not found';
+    for (const entry in report.libuv) {
+      if (report.libuv[entry].type === 'fs_event') fs = 1;
+      else if (report.libuv[entry].type === 'fs_poll') poll = 1;
+      else if (report.libuv[entry].type === 'process') process = 1;
+      else if (report.libuv[entry].type === 'timer') timer = 1;
+      else if (report.libuv[entry].type === 'pipe') pipe = 1;
+      else if (report.libuv[entry].type === 'tcp') tcp = 1;
+      else if (report.libuv[entry].type === 'udp') udp = 1;
+    }
+    assert.deepStrictEqual(fs, 1, fs_msg);
+    assert.deepStrictEqual(poll, 1, poll_msg);
+    assert.deepStrictEqual(process, 1, process_msg);
+    assert.deepStrictEqual(timer, 1, timer_msg);
+    assert.deepStrictEqual(pipe, 1, pipe_msg);
+    assert.deepStrictEqual(tcp, 1, tcp_msg);
+    assert.deepStrictEqual(udp, 1, udp_msg);
+
+    // Common report tests.
+    assert.ok(() => {
```

Comment:
`assert.ok()` doesn’t work, it just sees that it receives a function argument, which is truth-y